corehq/motech/fhir/repeater_helpers.py

```diff
@@ -28,13 +31,46 @@ def register_patients(
             # Patient is already registered
             info_resource_list_to_send.append((info, resource))
             continue
-        send_resource(requests, info, resource, repeater_id,
-                      raise_on_ext_id=True)
-        # Don't append `resource` to `info_resource_list_to_send`
-        # because the remote service has all its data now.
+        patient = find_patient(requests, resource)  # Raises DuplicateWarning
```

Comment:
It's funny to me that this is the client's responsibility to figure this out instead of the server that you are sending data to. 🤷🏽

Reply:
My introduction to this kind of thing (several years ago) was trying to match exam results with exam writers. The cover of the exam answer book came with instructions and a sample for exam writers to follow. The name in the sample was "Fezi Bam". ... Some writers wrote their name as "Fezi Bam". ... Sometimes servers end up with the darndest data!